Add resource dedicated_server_networking (#351)
ovh/terraform-provider-ovh: website/docs/r/ovh_dedicated_server_networking.html.markdown

---
layout: "ovh"
page_title: "OVH: ovh_dedicated_server_networking"
sidebar_current: "docs-ovh-resource-ovh-dedicated-server-networking"
description: |-
  Manage dedicated server networking interface
---

# ovh_dedicated_server_networking

Manage dedicated server networking interface.

## Example Usage

> The following example aims to bond all interfaces in vRack.

Reviewer: ```suggestion
The following example aims to bind all interfaces in a vRack
```

```hcl
locals {
  dedicated_server = "nsXXXXXX.ip-XX-XXX-XX.eu"
}

data "ovh_dedicated_server" "server" {
  service_name = local.dedicated_server
}

resource "ovh_dedicated_server_networking" "server"{
  service_name = local.dedicated_server
  interfaces {
      macs = sort(flatten(data.ovh_dedicated_server.server.vnis.*.nics)),
      type = "vrack"
  }
} 
```

The following example aims to attach the server to two different vRack.

```hcl
locals {
  dedicated_server = "nsXXXXXX.ip-XX-XXX-XX.eu"
}

data "ovh_dedicated_server" "server" {
  service_name = local.dedicated_server
}

> resource "ovh_dedicated_server_networking" "server"{

Reviewer: I have an error when trying to execute this example:

```
Terraform will perform the following actions:

  # ovh_dedicated_server_networking.server will be created
  + resource "ovh_dedicated_server_networking" "server" {
      + description  = (known after apply)
      + id           = (known after apply)
      + service_name = "ns<censored>.ip-xx-xx-xx.eu"
      + status       = (known after apply)

      + interfaces {
          + macs = [
              + "d0:xx:xx:xx:xx:b3",
              + "d0:xx:xx:xx:xx:b4",
            ]
          + type = "vrack"
        }
    }
    
 ...
    
ovh_dedicated_server_networking.server: Creating...
ovh_dedicated_server_networking.server: Still creating... [10s elapsed]
╷
│ Error: Error calling POST /dedicated/server/nsxxxxx.ip-xx-xx-xx.eu/networking:
│ 	 "HTTP Error 403: \"This account is not eligible to this API call\""
│
│   with ovh_dedicated_server_networking.server,
│   on pr351.tf line 11, in resource "ovh_dedicated_server_networking" "server":
│   11: resource "ovh_dedicated_server_networking" "server"{
```

Do you know why?

Thanks

Author: This is expected to be able to use it, for the moment, your nic must be allowed by baremetal team.
ping me when you want to test it as I have the necessary to test it with you.

  service_name = local.dedicated_server
  # Bond the two first interfaces in the first vrack
  interfaces {

>       macs = slice(sort(flatten(data.ovh_dedicated_server.server.vnis.*.nics)), 0, 2),

Reviewer: ```suggestion
      macs = slice(sort(flatten(data.ovh_dedicated_server.server.vnis.*.nics)), 0, 2)
```

      type = "vrack"
  }
  # Bond the two last interfaces in the last vrack
  interfaces {

>       macs = slice(sort(flatten(data.ovh_dedicated_server.server.vnis.*.nics)), 2, 4),

Reviewer: ```suggestion
      macs = slice(sort(flatten(data.ovh_dedicated_server.server.vnis.*.nics)), 2, 4)
```

>       macs = slice(sort(flatten(data.ovh_dedicated_server.server.vnis.*.nics)), 2, 4),

Reviewer: ```suggestion
      macs = slice(sort(flatten(data.ovh_dedicated_server.server.vnis.*.nics)), 2, 4)
```

      type = "vrack"
  }
} 
```

## Arguments Reference

The following arguments are required:

> * `service_name` - (String) The service_name of your dedicated server.

Reviewer: ```suggestion
* `service_name` - (String) The service_name of your dedicated server. The full list of available dedicated servers can be found using the `ovh_dedicated_servers` datasource.
```

* `interfaces` - (Block List, Min: 1, Max: 2) Interface or interfaces aggregation.

### Nested Schema for interfaces

Required:

> * `macs` (List of String) List of mac addresses to bond together.

Reviewer: ```suggestion
* `macs` (List of String) List of mac addresses to bind together.
```

* `type` (String) Type of bonding to create.

## Attributes Reference

The following attributes are exported:

* `description` - Operation description.

> * `status` - status of the networking configuration (should be `active`).

Reviewer: If importer have been implemented, please add it in the documentation also.
Example:
https://registry.terraform.io/providers/ovh/ovh/latest/docs/resources/cloud_project_kube#import

  service_name = local.dedicated_server
  # Bond the two first interfaces in the first vrack
  interfaces {
      macs = slice(sort(flatten(data.ovh_dedicated_server.server.vnis.*.nics)), 0, 2)
      type = "vrack"
  }
  # Bond the two last interfaces in the last vrack
  interfaces {
      macs = slice(sort(flatten(data.ovh_dedicated_server.server.vnis.*.nics)), 2, 4)
      type = "vrack"
  }
} 
```

## Arguments Reference

The following arguments are required:

* `service_name` - (String) The service_name of your dedicated server. The full list of available dedicated servers can be found using the `ovh_dedicated_servers` datasource.
* `interfaces` - (Block List, Min: 1, Max: 2) Interface or interfaces aggregation.

### Nested Schema for interfaces

Required:

* `macs` (List of String) List of mac addresses to bind together.
* `type` (String) Type of bonding to create.

## Attributes Reference

The following attributes are exported:

* `description` - Operation description.
* `status` - status of the networking configuration (should be `active`).

## Import

A dedicated server networking configuration can be imported using the `service_name`.

```bash

> $ terraform import 'ovh_dedicated_server_networking.server' service_name

Reviewer: ```suggestion
$ terraform import ovh_dedicated_server_networking.server service_name
```
